eclass/kernel-install.eclass

```diff
@@ -442,6 +595,41 @@ kernel-install_pkg_preinst() {
 	fi
 }
 
+# @FUNCTION: kernel-install_split_uki
+# @USAGE: <image path>
+# @DESCRIPTION:
+# Split a unified kernel image in the arch specific kernel image
+# and an initrd file. Does nothing unless KERNEL_IUSE_GENERIC_UKI is set.
+kernel-install_split_uki() {
+	[[ ${#} -eq 1 ]] || die "${FUNCNAME}: invalid arguments"
+	if [[ ${KERNEL_IUSE_GENERIC_UKI} ]]; then
```

Comment:
Leave this in postinst. Functions that only conditionally do something are confusing, and in ebuild *we know* whether we're setting it or not.